lib/reek/smells/smell_warning.rb

```diff
@@ -11,12 +11,14 @@ class SmellWarning
       attr_reader :context, :lines, :message, :parameters, :smell_detector
       def_delegators :smell_detector, :smell_category, :smell_type, :source
 
-      def initialize(smell_detector, options = {})
+      # FIXME: switch to required kwargs when dropping Ruby 2.0 compatibility
+      def initialize(smell_detector, context: '', lines: raise, message: raise,
+                                     parameters: {})
```

Comment:
:) I used to write the below (and do stuff like `lines: req(:lines)`) in my codebases that needed to support Ruby 2.0, but that was a bit too much. ;)

![req](https://cloud.githubusercontent.com/assets/56633/9517119/c4d8b1c0-4cad-11e5-9bb6-df82adaddafa.png)